apps/wallet-mobile/src/features/DappExplorer/useCases/WebBrowserScreen.tsx

```diff
@@ -0,0 +1,157 @@
+import {useFocusEffect} from '@react-navigation/native'
+import React, {useState} from 'react'
+import {Platform, Share, StyleSheet, TouchableOpacity, View} from 'react-native'
+import {WebView} from 'react-native-webview'
+
+import {TextInput} from '../../../components'
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useConnectWalletToWebView} from '../common/hooks'
+import {
+  NavigationArrowLeftIcon,
+  NavigationArrowRightIcon,
+  NavigationRefreshIcon,
+  NavigationShareIcon,
+} from '../common/icons'
+
+const DAPP_URL = 'https://app.dexhunter.io/'
+
+const ENABLED_BUTTON_COLOR = '#383E54'
+const DISABLED_BUTTON_COLOR = '#8A92A3'
+
+export const WebBrowserScreen = () => {
+  const [sessionId] = useState(() => Math.random().toString(36).substring(7))
+
+  return (
+    <View style={styles.root}>
+      <TabScreen sessionId={sessionId} />
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  root: {
+    flex: 1,
+  },
+  navigationContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    height: 32,
+    paddingVertical: 16,
+    paddingHorizontal: 7,
+  },
+  navigationButton: {
+    flex: 1,
+    width: 24,
+    height: 24,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+})
+
+function TabScreen({sessionId}: {sessionId: string}) {
```

Comment:
use Screen only for Modal first child and the component that takes over the screen.